packages/stream-model-instance-handler/src/model-instance-document-handler.ts

```diff
@@ -228,7 +228,7 @@ export class ModelInstanceDocumentHandler implements StreamHandler<ModelInstance
    * @param header - the header to validate
    */
   async _validateHeader(model: Model, header: ModelInstanceDocumentHeader): Promise<void> {
-    if (model.content.accountRelation == ModelAccountRelation.SINGLE) {
+    if (model.content.accountRelation.type === 'single') {
```

Comment:
I didn't totally follow the whole discussion above about enums and bundle size, but can we have `ModelAccountRelationType` string constants so we're not using raw strings here?  Makes it clearer to readers of the code that this isn't an arbitrary string field but has a pre-defined set of supported values that need to be considered.

Reply:
That's what the `ModelAccountRelation` type ensures, here the `type` can't be any value, the TS compiler will throw if it's not `single` or `list`.

Reply:
sure, but it's about more than the compiler catching errors, it's about code legibility.  When I see a raw string constant I wonder "what else can this be?  Is this string user-provided?  Do I need to handle arbitrary input?".  When I see a namespaced constant like `ModelAccountRelationType.SINGLE` - I know immediately that we're dealing with a fixed set of supported values

Reply:
Yeah that's the conversation we had with Sergey in another comment above, the problem with enums is that they are not "just types", they are actual objects, and so to use them in consuming packages, we need to import the actual model package rather than just the types.
It makes the runtime/production dependency tree heavier than it needs to be, for no additional type safety, to me the cost is really not worth the benefits, but if you feel strongly about it I think we should discuss other alternatives, such as a dedicated package that only contains such constants so we don't need to import the rest of the code?

Reply:
yeah, at some point in the future maybe having a separate "constants" package would make sense, probably not worth it right now though.  I do still prefer the enum approach for legibility, but I also don't know how big of an impact on bundle size it actually makes in practice.  In any case I don't feel so strongly about this - if you think the bundle size difference is meaningful then I'm fine to leave this as-is

Reply:
We're not tracking bundle size yet so I don't know either, the problem I see is that by the time we get there, we'll probably already have shipped a stable version of these packages and we won't be able to make these kinds of breaking changes anymore without having to bump a major version.
If you think we can get to the point of having a separate constants package by the time we release a first stable version of the model package I think it's OK, but otherwise better use types than enums I think.